homeassistant/components/sensor/fido.py

```diff
@@ -0,0 +1,435 @@
+USAGE_URL = "{}/v1/postpaid/dashboard/usage".format(HOST_FIDO)
+
+
+DATA_MAP = {'data': ('data', 'D'),
+            'text': ('text', 'BL'),
+            'mms': ('text', 'M'),
+            'text_int': ('text', 'SI'),
+            'talk': ('talk', 'V'),
+            'talk_other': ('talk', 'VL')}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fido sensor."""
+    # Create a data fetcher to support all of the configured sensors. Then make
+    # the first call to init the data.
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        fido_data = FidoData(username, password)
+        fido_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(error)
+        return False
+
+    name = config.get(CONF_NAME)
+
+    sensors = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        sensors.append(FidoSensor(fido_data, variable, name))
+
+    add_devices(sensors)
+
+
+class FidoSensor(Entity):
+    """Implementation of a Fido sensor."""
+
+    def __init__(self, fido_data, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.type = sensor_type
+        self.entity_id = "sensor.{}_{}".format(name, sensor_type)
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self.fido_data = fido_data
+        self._state = None
+
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    def update(self):
+        """Get the latest data from Fido and update the state."""
+        self.fido_data.update()
+        if self.type in self.fido_data.data:
+            if self.fido_data.data[self.type] is not None:
+                self._state = round(self.fido_data.data[self.type], 2)
+
+
+class FidoData(object):
+    """Get data from HydroQuebec."""
+
+    def __init__(self, number, password):
+        """Initialize the data object."""
+        self.number = number
+        self.password = password
+        self.data = {}
+        self.headers = {'User-Agent': ('Mozilla/5.0 (X11; Linux x86_64; '
+                                       'rv:10.0.7) Gecko/20100101 '
+                                       'Firefox/10.0.7 Iceweasel/10.0.7')}
+        self.cookies = None
+
+    def _post_login_page(self):
```

Comment:
There is too much protocol specific things in this file. Please extract it into a third party lib that this platform can depend on.